more_itertools/more.py

```diff
@@ -180,6 +181,23 @@ def last(iterable, default=_marker):
         return default
 
 
+def nth_or_last(iterable, n, default=_marker):
+    """Return the nth or the last item of *iterable*,
+    or *default* if *iterable* is empty.
+
+        >>> nth_or_last([0, 1, 2, 3], 2)
+        2
+        >>> nth_or_last([0, 1], 2)
+        1
+        >>> nth_or_last([], 0, 'some default')
+        'some default'
+
+    If *default* is not provided and there are no items in the iterable,
+    raise ``ValueError``.
+    """
+    return last(islice(iterable, n + 1), default=default)
```

Comment:
Clever implementation!  `islice` yields the full elements even if `n` > `len(seq)`, so you are guaranteed to get the last element.  Nice.